src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -57,13 +57,30 @@ public static Process Start(string fileName, string arguments, string userName,
         /// <summary>Terminates the associated process immediately.</summary>
         public void Kill()
         {
-            EnsureState(State.HaveNonExitedId);
-            if (Interop.Sys.Kill(_processId, Interop.Sys.Signals.SIGKILL) != 0)
+            EnsureState(State.HaveId);
+
+            // Don't do anything if we know the process has exited.
+            // This also protects us from killing another process with a recycled PID.
+            if (GetHasExited(refresh: false))
```

Comment:
Why `refresh: false` rather than `refresh: true`?

Reply:
This is a small optimization. Just below we make a `kill` call that will actively check if the process still exists, so here we can just check our _internal state_ without making a system call.

Reply:
The "This also protects us..." part makes it sounds like it's more than just an optimization.

Reply:
It's optimized compared to calling 'HasExited'. We need to check exited here for pid recycling. `EnsureState(State.HaveNonExitedId);` was also doing `refresh: false`.

I'll improve the comment.